### ISSUE TITLE
Autocomplete Pollution

### ISSUE THREAD

**Issue by NonSpicyBurrito** [fabian-hiller/valibot#932]
In VS Code, when you type `string`, VS Code autocomplete suggests `import { string } from 'valibot'` which is undesirable. This can be turned off by using `^valibot$` pattern in `typescript.preferences.autoImportSpecifierExcludeRegexes`.

However, if you have Valibot imported like so:
```ts
import * as v from 'valibot'
```
When you type `string`, VS Code autocomplete suggests "change `string` to `v.string`" which is also undesirable.

This not only happens to `string`, but also `object`, `date`, `void`, `undefined`, `RegExp` (due to `v.regex`), `await` (due to `v.awaitAsync`), `return` (due to `v.returns`) etc. This makes typing code like `Date.now()` via `date[tab to autocomplete to Date].now()` unusable as tab autocompletes to `v.date` instead, and also makes using your own snippets for keywords unusable (for example I have `ret` snippet which expands to `return $0`).

**Comment by fabian-hiller** [comment]
Do you know if there is an option to disable import completion for a specific library in VS Code?

**Comment by NonSpicyBurrito** [comment]
Yeah `autoImportSpecifierExcludeRegexes` only disables imports, and "change `string` to `v.string`" does not seem to be an import.

**Comment by NonSpicyBurrito** [comment]
My current work around is to isolate all Valibot related code in their own modules, which is not really ideal. I have schemas in their own files, and re-export things like `v.is`.

**Comment by fabian-hiller** [comment]
I don't have the "change `string` to `v.string`" problem in my VS Code. Can you share a screenshot?

**Comment by NonSpicyBurrito** [comment]
![valibot](https://github.com/user-attachments/assets/fbe65260-2613-4d75-8aaf-4c09fb647f0e)

**Comment by fabian-hiller** [comment]
In my VS Code the ordering is different:

![Screenshot](https://github.com/user-attachments/assets/87ecbd03-0caf-4298-be86-9e8971caaa54)

**Comment by fabian-hiller** [comment]
Do you have any idea what we could do to improve this? Without a VS Code extension, we as a library probably have no control over the behaviour of the editor.

**Comment by NonSpicyBurrito** [comment]
I'm unsure. Some ideas:
- Rename to avoid collision, eg `vString` instead of `string`. This however would be a giant breaking change so it might be unrealistic.
- From my understanding the "change `string` to `v.string`" autocomplete comes from TS language server, so perhaps a feature request for adding exclude patterns similar to import exclude patterns.

**Comment by andersk** [comment]
Note that VS Code gives these completions in value context but not in type context:

```ts
const a = string // suggests v.string
const b: string // suggests string
```

I don’t see anything wrong with this behavior? `string` is not ordinarily valid in value context anyway. If you’ve declared a variable named `string`, then `string` is suggested before `v.string`, and similarly for an identifier like `undefined` that’s valid in value context.

It’s true that `date` suggests `v.date` before `Date`, but `Date` suggests `Date` before `v.date`, so that hardly qualifies as “unusable”.

**Comment by NonSpicyBurrito** [comment]
It cause specifically "typing lowercase `date` then auto completes to `Date`" unusable (as well as things like `reg` -> `RegExp`). It is correct that typing uppercase `Date` still works, but that's a different way of typing. It may not affect you, but to others, bringing in a library causes you to have to type in a different way is not ideal.

**Comment by miyaji255** [comment]
> Rename to avoid collision, eg vString instead of string. This however would be a giant breaking change so it might be unrealistic.

`import * as v from "valibot";` cannot be auto-completed. I believe this issue can also be resolved by using a prefix like `vString`.
